src/script/event/preprocessor/QuoteDecoderMiddleware.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {Quote} from '@wireapp/protocol-messaging';
+
+import {MessageAddEvent} from 'src/script/conversation/EventBuilder';
+import {getLogger, Logger} from 'Util/Logger';
+import {base64ToArray} from 'Util/util';
+
+import {QuoteEntity} from '../../message/QuoteEntity';
+import {StoredEvent} from '../../storage/record/EventRecord';
+import {ClientEvent} from '../Client';
+import {EventMiddleware, IncomingEvent} from '../EventProcessor';
+import type {EventService} from '../EventService';
+
+export class QuotedMessageMiddleware implements EventMiddleware {
```

Comment:
This entire class is a copy/paste of method from the original `QuotedMessageMIddleware`. Code can be overlooked